ossl-shim.patch

```diff
@@ -1057,3 +1068,403 @@ index dfbd380..75b03f2 100644
          fprintf(stderr, "Invalid base64: %s\n", argv[i]);
          return false;
        }
+-- 
+2.7.4
+
+
+From 1fd30c3bf39c95555464d9cf52e1436e16c9faa8 Mon Sep 17 00:00:00 2001
+From: Matt Caswell <matt@openssl.org>
+Date: Tue, 21 Jun 2016 12:06:26 +0100
+Subject: [PATCH 2/2] Update bssl_shim for some opaque changes
+
+BIO, DH and STORE_CTX were made opaque. These changes enable bssl_shim to
+build against the latest master version
+---
+ ssl/test/async_bio.cc    | 72 +++++++++++++++++++++++++++---------------------
+ ssl/test/bssl_shim.cc    | 29 ++++++++++++++-----
+ ssl/test/packeted_bio.cc | 63 ++++++++++++++++++++++--------------------
+ 3 files changed, 95 insertions(+), 69 deletions(-)
+
+diff --git a/ssl/test/async_bio.cc b/ssl/test/async_bio.cc
+index 965a0fb..825b223 100644
+--- a/ssl/test/async_bio.cc
++++ b/ssl/test/async_bio.cc
+@@ -22,8 +22,6 @@
+ 
+ namespace {
+ 
+-extern BIO_METHOD g_async_bio_method;
+-
+ struct AsyncBio {
+   bool datagram;
+   bool enforce_write_quota;
+@@ -32,20 +30,26 @@ struct AsyncBio {
+ };
+ 
+ AsyncBio *GetData(BIO *bio) {
++  /*
++   * TODO: Missing accessor? This probably needs a BIO_get_method() in OpenSSL
```

Comment:
Ah, not important (it's just a sanity check, right?). Rest looks ok as far as reviewing a patch diff is possible... Not sure what's going on with the hangs but I'll merge this first and let's take it from there!